tests/acl/custom_acl_table/test_custom_acl_table.py

```diff
@@ -0,0 +1,265 @@
+import logging
+import json
+import pytest
+import time
+
+from ptf.mask import Mask
+import ptf.packet as scapy
+
+
+import ptf.testutils as testutils
+from tests.common.helpers.assertions import pytest_assert
+from tests.common.plugins.loganalyzer.loganalyzer import LogAnalyzer, LogAnalyzerError
+from tests.common.dualtor.mux_simulator_control import toggle_all_simulator_ports_to_rand_selected_tor
+
+logger = logging.getLogger(__name__)
+
+pytestmark = [
+    pytest.mark.topology("t0"), # Only run on T0 testbed
+    pytest.mark.disable_loganalyzer,  # Disable automatic loganalyzer, since we use it for the test
+]
+
+CUSTOM_ACL_TABLE_TYPE_SRC_FILE = "acl/custom_acl_table/custom_acl_table.json"
+CUSTOM_ACL_TABLE_TYPE_DST_FILE = "/tmp/custom_acl_table.json"
+
+ACL_RULE_SRC_FILE = "acl/custom_acl_table/acl_rules.json"
+ACL_RULE_DST_FILE = "/tmp/acl_rules.json"
+
+LOG_EXPECT_ACL_TABLE_CREATE_RE = ".*Created ACL table.*"
+LOG_EXPECT_ACL_RULE_FAILED_RE = ".*Failed to create ACL rule.*"
+
+
+@pytest.fixture(scope='module')
+def setup_counterpoll_interval(rand_selected_dut):
+    """
+    Set the counterpoll interval for acl to 1 second (10 seconds by default)
+    """
+    # Set polling interval to 1 second
+    rand_selected_dut.shell('counterpoll acl interval 1000')
+    time.sleep(10)
+    yield
+    # Restore default value 10 seconds
+    rand_selected_dut.shell('counterpoll acl interval 1000')
+
+
+def clear_acl_counter(dut):
+    """
+    Clear the counter of ACL
+    """
+    dut.shell('aclshow -c')
+
+
+def read_acl_counter(dut, rule_name):
+    """
+    Read the counter of given rule
+    RULE NAME    TABLE NAME      PRIO    PACKETS COUNT    BYTES COUNT
+    -----------  ------------  ------  ---------------  -------------
+    RULE_1       L3_MIX_TABLE    9999                0              0
+    """
+    cmd = 'aclshow -a -r {}'.format(rule_name)
```

Comment:
Maybe we could use `show_and_parse` here: https://github.com/sonic-net/sonic-mgmt/blob/master/docs/api_wiki/sonichost_methods/show_and_parse.md

Reply:
Updated. Thanks